backend/src/main/java/ca/mcgill/purposeful/controller/ReactionController.java

```diff
@@ -0,0 +1,55 @@
+package ca.mcgill.purposeful.controller;
+
+import ca.mcgill.purposeful.dto.ReactionDTO;
+import ca.mcgill.purposeful.exception.GlobalException;
+import ca.mcgill.purposeful.model.Reaction;
+import ca.mcgill.purposeful.model.Reaction.ReactionType;
+import ca.mcgill.purposeful.service.ReactionService;
+import java.util.Date;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * API for accessing the endpoints of Reaction
+ */
+@RestController
+@RequestMapping({"/api/reaction", "/api/reaction/"})
+public class ReactionController {
+
+  @Autowired
+  private ReactionService reactionService;
+
+  /**
+   * POST method to react
+   *
+   * @param reactionDTO - the reaction added or removed
+   * @return the newly created reaction DTO or the removed reaction DTO with null values
+   * @author Athmane Benarous
+   */
+  @PostMapping(value = {"/react", "/react/"})
+  @PreAuthorize("hasAnyAuthority('User', 'Moderator', 'Owner')")
+  public ResponseEntity<ReactionDTO> react(@RequestBody ReactionDTO reactionDTO) {
+    // Unpack the DTO
+    if (reactionDTO == null) {
+      throw new GlobalException(HttpStatus.BAD_REQUEST, "reactionDTO is null");
+    }
+    Date date = reactionDTO.getDate();
```

Comment:
Do we really need to pass a date to create a reaction? What do you think about just generating the current date from the service?

Reply:
We may not need to pass it, but it may be necessary as part of the DTO.
Great point, however. Instead of reactionDTO.getDate(), I will pass the current date into the returned DTO.

Reply:
Done!